internal/plan/plan.go

```diff
@@ -63,10 +80,14 @@ func (p PlanHandler) ReadPlan() (OptimisationPlan, error) {
 }
 
 func (p PlanHandler) WritePlan(optPlan OptimisationPlan) error {
+	p.mu.Lock()
```

Comment:
a mutex helps us when we deal with concurrency and race conditions.
Why does the planner needs to worry about that? couldn't the planner be thought as a single threaded, non-concurrent type of application?

Reply:
Mind due, I am not implying what you have done here is wrong at all. I am just asking to learn why we need this.

Reply:
I thought it would be safer to lock and unlock the file given that it's been written and read by different parts of the system - the MQTT subscriber callback `handlePlanMessage` writes the plan to file, and `CheckForOutage` reads it from the ticker-based goroutine (and also rewrites it, if we keep the `TrimPlan` part)

Reply:
👍 probably makes sense then